src/services/telegram/commands/roll.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{Context};
+use rand::{Rng, SeedableRng};
+use teloxide::Bot;
+use teloxide::payloads::SendMessageSetters;
+use teloxide::prelude::{Message, Requester};
+use crate::services::telegram::commands::{CommandResult};
+use regex::Regex;
+
+
+
+pub async fn handler(bot: &Bot, message: &Message, roll: &str) -> CommandResult {
+    let mut rng = rand::rngs::SmallRng::from_entropy();
+    if rng.gen_range(1..1001) == 1 {
+        let _reply = bot
+		.send_message(message.chat.id, "🎶 Roll? Rick roll! https://www.youtube.com/watch?v=dQw4w9WgXcQ")
+		.reply_to_message_id(message.id)
+		.await
+		.context("Non è stato possibile inviare la risposta.")?;
+
+	    return Ok(())
+    }
+
+    let re = Regex::new(r#"(?P<qty>[0-9]*)?d(?P<die>[0-9]+)(?P<modifier>[+-]?[0-9]*)?"#);
+    let mut qty = 1;
+    let mut die = 0;
+    let mut modifier = 0;
+
+    match re?.captures(roll) {
+        Some(caps) => {
+            qty = caps["qty"].parse().unwrap_or(qty);
+            die = caps["die"].parse().unwrap_or(die);
+            modifier = caps["modifier"].parse().unwrap_or(modifier);
+        }
+        None => {}
+    }
+
+    if die <= 0  {
+        let _reply = bot
+		.send_message(message.chat.id, "Specificare almeno un dado.")
+		.reply_to_message_id(message.id)
+		.await
+		.context("Dado = 0")?;
+
+	    return Ok(())
+    }
+
+    if qty < 1  {
+        let _reply = bot
+		.send_message(message.chat.id, "La quantità di dadi da tirare deve essere un intero positivo. (lasciare vuoto per sottintendere 1)")
+		.reply_to_message_id(message.id)
+		.await
+		.context("Qty < 1")?;
+
+	    return Ok(())
+    }
+
+
+
+    let mut nums_rolled = Vec::<i32>::new();
+    for _i in 0..qty {
```

Comment:
Se ricordo bene, possiamo direttamente ignorare l'indice del dado:

```suggestion
    for _ in 0..qty {
```